cmake/app/boilerplate.cmake

```diff
@@ -27,6 +27,12 @@ cmake_minimum_required(VERSION 3.8.2)
 
 cmake_policy(SET CMP0002 NEW)
 
+if(NOT (${CMAKE_VERSION} VERSION_LESS "3.13.0"))
+  # Use the old CMake behaviour until build scripts have been ported
+  # to the new behaviour.
```

Comment:
Just a small comment. Using cmake policy CMP0079 will still work with current cmake files, as CMP0079 allows to use target_link_libraries from any directory.
https://cmake.org/cmake/help/latest/policy/CMP0079.html

The reason for setting CMP0079 to `OLD` is to ensure Zephyr's CMake tree is still supporting older CMake versions, i.e. 3.8
So comment should be something like:
```
# Use old CMake behavior as minimum cmake required version in Zephyr is 3.8.2, and versions prior to 3.13 does not support new behavior of CMP0079
```

Reply:
AFAICT NEW does not (necessarily) work with the current cmake files. I get this from here:

> It did accidentally accept targets from other directories on calls that only update the INTERFACE_LINK_LIBRARIES, but would simply add entries to the property as if the call were made in the original directory. Thus link interface libraries specified this way were always looked up by generators in the scope of the original target rather than in the scope that called target_link_libraries().

> This policy provides compatibility with projects that already use target_link_libraries() with the INTERFACE keyword on a target in another directory to add INTERFACE_LINK_LIBRARIES entries to be looked up in the target’s directory. Such projects should be updated to be aware of the new scoping rules in that case.

I could be misreading this, it is difficult to read. But it seems to me that changing from OLD to NEW will change the scope will change in some cases.

I'll add a note about this also being dependent on using 3.13.

Reply:
Thank you for the in-depth analysis, your description is much better than the official documentation ;)